### PR TITLE
Avatars from url

### PR DESCRIPTION
**Pull request by graywolf336**
This allows the avatars to be set from a provided url, the url must return a type of jpg or else it isn't valid. This is needed for my slack import so that I can set the users avatars from slack's. This does import it into the avatar filestore so that they're "cached".

### CODE REVIEW

**Comment by graywolf336** [comment]
I will admit, the url box and the submit button aren't very nice looking...but I couldn't figure out how to get it to look right with my limited knowledge of the css for the project. If the styling of this is the only thing holding back, then let me know and I can try to do something with it.

**Comment by geekgonecrazy** [comment]
Would it look better without the upload icon on the left side?

**Comment by graywolf336** [comment]
Maybe? But not sure, as the rest of the services have an icon next to them too

File: .meteor/packages [hunk 1]
```diff
@@ -109,4 +109,5 @@ yasaricli:slugify
 yasinuslu:blaze-meta
 # sanjo:jasmine
 # velocity:html-reporter
+froatsnook:request
```
Comment (rodrigok) [inline .meteor/packages]: Hey @graywolf336, why you are using this link instead the internal HTTP lib?

Comment (graywolf336) [inline .meteor/packages]: I've used the npm module this is based off before, I wasn't aware of an http lib. I will check that out and then update this pull request. @rodrigok

Comment (rodrigok) [inline .meteor/packages]: :smile: take a look here https://github.com/RocketChat/Rocket.Chat/blob/master/server/methods/getAvatarSuggestion.coffee#L48

**Comment by graywolf336** [comment]
@rodrigok How's it look now?

File: server/methods/setAvatarFromService.coffee [hunk 2]
```diff
@@ -11,6 +11,28 @@ Meteor.methods
...
 			return
 
+		if service is 'url'
+			try
+				result = HTTP.get dataURI, npmRequestOptions: {encoding: 'binary'}
+
+				if result.statusCode isnt 200
+					console.log "Not a valid response, #{result.statusCode}, from the avatar url: #{dataURI}"
+					throw new Meteor.Error('invalid-avatar-url', '[methods] setAvatarFromService -> url service -> error on getting the avatar from url')
+
```
Comment (rodrigok) [inline server/methods/setAvatarFromService.coffee]: Aren't you checking the type of result ('image/jpeg') anymore?

Comment (graywolf336) [inline server/methods/setAvatarFromService.coffee]: I decided against it since the avatars can be gifs, pngs, or jpgs. Should I add the checks for those?

Comment (rodrigok) [inline server/methods/setAvatarFromService.coffee]: Sure, but IMHO you should check for the first part of the data type `/image\/.+/` at least

Comment (graywolf336) [inline server/methods/setAvatarFromService.coffee]: Then that I will do, thanks. Another problem I just noticed is the error isn't presented back to the user if it occurs..

Comment (rodrigok) [inline server/methods/setAvatarFromService.coffee]: Can you open another issue for this? We need to improve how to show errors at all.

Comment (graywolf336) [inline server/methods/setAvatarFromService.coffee]: Done, #1273.

**Comment by engelgabriel** [comment]
So are we going to merge this one and do another PR for the check for the first part of the data type? Or should I wait for this PR to be updated?

**Comment by graywolf336** [comment]
I'm fine with either of those options. I won't be able to push the checks
until tomorrow.

**Comment by engelgabriel** [comment]
Let's merge and you do the other issue tomorrow.

**Pull request merged**